backend/user/user.py

```diff
@@ -13,61 +13,84 @@ def checkIn(self, id_rent):
 
         current_date = datetime.date.today()
 
-        sql = f"SELECT * FROM rents WHERE rentId = '{id_rent}'"
+        preCheckIn = f"SELECT bookingStart FROM rents WHERE rentId = '{id_rent}'"
 
-        cursor.execute(sql)
+        cursor.execute(preCheckIn)
+
+        if preCheckIn == current_date:
+
+            sql = f"SELECT * FROM rents WHERE rentId = '{id_rent}'"
 
-        update = (
-        f"UPDATE rents"
-        f"SET checkIn = '{current_date}'"
-        f"WHERE rentId = '{id_rent}'")
+             cursor.execute(sql)
 
-        cursor.execute(update)
+             update = (
+             f"UPDATE rents"
+             f"SET checkIn = '{current_date}'"
+             f"WHERE rentId = '{id_rent}'")
 
-        validation = f"SELECT checkIn FROM rents WHERE rentId = '{id_rent}'"
+             cursor.execute(update)
 
-        cursor.execute(validation)
+            validation = f"SELECT checkIn FROM rents WHERE rentId = '{id_rent}'"
+
+            cursor.execute(validation)
+
+            if validation != current_date:
+                return {
+                    'success': False,
+                    'error': 'Unable to checkin'
+                }
 
-        if validation != current_date:
             return {
-                'success': False,
-                'error': 'Unable to checkin'
+                'success': True
             }
 
         return {
-                'success': True
-        }
+                'success': False,
+                'error': 'Booking date not found.'
```

Comment:
Modifica a mensagem de erro pra algo q avise q o usuário só pode dar check-in no dia da reserva